lib/foreman_maintain/utils/sync_plans.rb

```diff
@@ -0,0 +1,112 @@
+module ForemanMaintain
+  module Utils
+    module SyncPlans
```

Comment:
I suggest extracting the code regarding the yaml serialization into a 'persistence' object, that will handle the manipulation of yaml (loading/saving), and trying to extract the hammer-specific code into hammer feature, leaving there only the code specific to the  sync plans.